history/history-network.md

```diff
@@ -165,9 +165,25 @@ each receipt/transaction and re-rlp-encode it, but only if it is a legacy transa
 ```python
 # Content types
 
+# Proof for EL BlockHeader before TheMerge / Paris
 HistoricalHashesAccumulatorProof = Vector[Bytes32, 15]
 
-BlockHeaderProof = Union[None, HistoricalHashesAccumulatorProof]
+# Proof that EL block_hash is in  BeaconBlock -> BeaconBlockBody -> ExecutionPayload
+BeaconBlockProof = Vector[Bytes32, 11]
+
+# Proof that BeaconBlock root is part of HistoricalRoots and thus canonical
+# For after TheMerge until Capella -> Bellatrix fork.
+HistoricalRootsProof = Vector[Bytes32, 14]
+
+# Proof for EL BlockHeader after TheMerge until Capella
+HistoricalRootsBlockProof = Container[
+    beaconBlockProof: BeaconBlockProof,
+    beaconBlocRoot: Bytes32,
```

Comment:
Is the the beacon block that the proof is anchored to?

Reply:
As I mentioned here in this link,  https://github.com/ethereum/portal-network-specs/pull/287#discussion_r1569186217, would it be more clear to go with object naming where  the name of the proof is about what it proves (instead of "how")?

```py
BlockProofHistoricalRoots = Container[
    beaconBlockProof: BeaconBlockProofHistoricalRoots, # Proof that the BeaconBlock is part of the historical_roots and thus part of the canonical chain
    beaconBlockRoot: Bytes32,
    executionBlockProof: ExecutionBlockProof, # Proof that EL BlockHash is part of the BeaconBlock
    slot: Slot
]
```

The reason I still attach the `HistoricalRoots` part at the end, is because there is also the `HistoricalSummaries` version. This would then be:

```py
BlockProofHistoricalSummaries = Container[
    beaconBlockProof: BeaconBlockProofHistoricalSummaries, # Proof that the BeaconBlock is part of the historical_summaries and thus part of the canonical chain
    beaconBlockRoot: Bytes32,
    executionBlockProof: ExecutionBlockProof, # Proof that EL BlockHash is part of the BeaconBlock
    slot: Slot
]
```